pimd/pim_zlookup.c

```diff
@@ -125,7 +125,9 @@ static void zclient_lookup_failed(struct zclient *zlookup)
 
 void zclient_lookup_free(void)
 {
-	THREAD_OFF(zlookup_read);
+	if (zlookup_read)
```

Comment:
Can you explain what this fixes - when I look at `THREAD_OFF`, it appears to be safe to call with NULL?